project_task_stage_change_restriction/models/project_task.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    def _is_move_allowed(self, task, new_stage, user):
+        """Return True if **user** may move **task** into **new_stage** (OR-logic).
```

Comment:
How many times should it be reminded that complete docstings are mandatory for ALL functions? 😄